server/app/routes/users.js

```diff
@@ -46,75 +47,43 @@ router.get('/:userId/getSwag', ensureAuthenticated, function(req, res, next) {
     User.findOne({ _id: req.params.userId })
     .populate('animal.swags')
     .then( user => {
-        res.json(user)})
+        res.json(user);
+    })
+    .then(null, next);
 });
 
 router.put('/:userId/getSwag/:swagId', ensureAuthenticated, function (req, res, next) {
-    if (!req.user) return 'User not found!'
-    Swag.findOne({ _id: req.params.swagId })
-    .then(function(swag) {
-        req.user.animal.swags.push(swag);
-        req.user.animal.money = req.user.animal.money - swag.price;
+    if (!req.user) return 'User not found!';
+    var swagToBuy = req.body;
+    Swag.create({
+        name: swagToBuy.name,
+        crategory: swagToBuy.crategory,
+        price: swagToBuy.price,
+        imageUrl: swagToBuy.imageUrl
+        })
+    .then(function(createdSwag) {
+        req.user.animal.swags.push(createdSwag);
+        req.user.animal.money = req.user.animal.money - createdSwag.price;
         return req.user.save();
```

Comment:
haha, that spelling error killed me for a while, it should be okay now. I think we'll need to add something to the swag model so that we don't display every swag ever created in the store, because this does create a new instance of the swag right now, and when we do a get request for all swag, I think it's also including the things that are being purchased....